### PR TITLE
Delete rand.py

### LINKED ISSUE #674: deprecation warning: OpenSSL.rand, despite not importing it
I've got some CI tests that look for deprecation warnings (in the py27 build), and these have recently started complaining about something internal in Twisted's use of OpenSSL.

Twisted (in src/twisted/protocols/tls.py) does `from OpenSSL.SSL import (somestuff)`, and `OpenSSL/__init__.py` does `from OpenSSL import rand`, and `rand.py` has a top-level `warnings.warn` that says `rand` is deprecated and applications should use `os.urandom` instead (added for #631).

```
% PYTHONWARNINGS=default::DeprecationWarning python -c "import OpenSSL"
-c:1: DeprecationWarning: OpenSSL.rand is deprecated - you should use os.urandom instead
%
```

That advice is fine, of course, but I don't see a way for applications to avoid triggering the import-time warning (Twisted isn't using or even importing `OpenSSL.rand`).

If apps are depending upon being able to do `import OpenSSL; OpenSSL.rand.bytes()` (rather than `from OpenSSL.rand import bytes`), then I guess we can't remove the import from `__init__.py`. Could we maybe move the warning calls to the individual functions?

### PR DESCRIPTION
**Pull request by alex**
Fixes #674

### CODE REVIEW

**Comment by alex** [comment]
Failing tests are because of:

- https://github.com/twisted/twisted/pull/847
- #676

**Comment by hynek** [comment]
Well, that’s kind of a violation of our backward compatibility policy, isn’t it?  I’m not saying no, but we should at least try to explain why we do it?

Als we should move to incremental. *eyes merge conflict*

**Comment by alex** [comment]
I guess it technically is. The fact that as far as I can tell, _no one_ uses this is a serious mitigation in my mind though.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,8 @@ Backward-incompatible changes:
 
 - Dropped support for Python 3.3.
   `#677 <https://github.com/pyca/pyopenssl/pull/677>`_
+- Removed the deprecated ``OpenSSL.rand`` module. ``os.urandom`` should be used instead.
```
Comment (hynek) [inline CHANGELOG.rst]: Please:

1. add a rationale why we choose to disregard our own policy (1 sentence is just fine)
2. make it `os.urandom()` – marking functions/methods with parentheses is nicer to read/understand.

thanks!

**Comment by hynek** [comment]
Making the world a safer place.  One `git rm` a time.

**Comment by tbaxx** [comment]
@hynek, @alex, using urandom is not an alternative for the whole rand module, just for some functions like rand.bytes.

There are other functions in the module, such as rand.add that are changing the PRNG state of OpenSSL. Using such functions is of particular importance for fork safety, as described here: https://wiki.openssl.org/index.php/Random_fork-safety

Kindly note that the link states that "avoiding RAND_bytes is not practical in practice because the library will use it internally". Therefore, OpenSSL *will* use rand.bytes (whether it is present in pyopenssl or not) and that's why applications that are forking (e.g. when using the multiprocessing module in Linux) will need to make sure the PRNG state of OpenSSL is different between the parent and child process.

So, I don't see how you are making the world a safer place with this change, I think you are doing the opposite.

Please consider reverting this change and finding another way to fix any headaches caused by the presence of the rand module.

**Pull request merged**